src/datasets/features/features.py

```diff
@@ -1698,14 +1701,14 @@ def unsimplify(feature: dict) -> dict:
             #
             # class_label:              ->              class_label:
             #   names:                  ->                names:
-            #     0: negative              ->               - negative
-            #     1: positive              ->               - positive
+            #     '0': negative              ->               - negative
+            #     '1': positive              ->               - positive
             #
             if isinstance(feature.get("class_label"), dict) and isinstance(feature["class_label"].get("names"), dict):
                 label_ids = sorted(feature["class_label"]["names"])
```

Comment:
because "2" > "10" in python
```suggestion
                label_ids = sorted([int(label_id) for label_id in feature["class_label"]["names"]])
```

Reply:
I agree that the sort should be fixed for strings.

However, I wouldn't transform `label_ids` to `int` here: see some lines below:
https://github.com/huggingface/datasets/blob/2205b5f5135d0c36f9a3527beebea0aa9ee89901/src/datasets/features/features.py#L1713

This would raise an error:
- `label_id` is int
- but keys of `feature["class_label"]["names"]` are str

Reply:
I have used instead:
```python
label_ids = sorted(feature["class_label"]["names"], key=int)
```

Reply:
Ok ! Feel free to merge when it's all good to you